### ISSUE TITLE
Reuse strategic merge patches for multiple targets

### ISSUE THREAD

**Issue by hyww** [kubernetes-sigs/kustomize#638]
It seems that we cannot reuse strategic merge patches for multiple targets since targets are defined inside each patch.

I've come up with an idea that maybe we can put strategic merge patches in `List` and reuse duplicated part  of patches using anchor/reference of YAML.

#637 is my implementation and below is an example to apply `hostAliases` patch to multiple `Deployment`s

```
apiVersion: v1
kind: List
items:
- apiVersion: apps/v1
  kind: Deployment
  metadata:
    name: deployment-a
  spec: &hostAliases
    template:
      spec:
        hostAliases:
        - hostnames:
          - a.example.com
          ip: 1.1.1.1
        - hostnames:
          - b.example.com
          ip: 8.8.8.8
- apiVersion: apps/v1
  kind: Deployment
  metadata:
    name: deployment-b
  spec:
    <<: *hostAliases
```

If this seems to be a correct method to deal with the problem, I might be able to work on this further and provide tests, documentations, examples, etc.

**Comment by Liujingfang1** [comment]
This is another issue for `List`, which is to load resources from a `List` #581

**Comment by Liujingfang1** [comment]
fixed by #637
